src/components/DeckMechanisms/index.js

```diff
@@ -158,10 +159,56 @@ export default class DeckMechanisms extends React.Component {
 
         // Remove the played card from the hand.
         newState.hand = state.hand.filter(cardId => cardId !== id)
-
+        
         if (options.discard) {
           return newState
         }
+        
+        // Turn one: Check if board is not full (by Rain of Frogs for example)
+        if (newState.turn === 1) {
+          const cardData = resolveCardForLevel(card)
+          // Any 3 mana card can be played since it would be the only one to be played
+          // and would not fill the board by itself. Collector Mirz creating a 0 mana
+          // token is not an issue for board filling.
+
+          // Any 2 mana card will have to be played together with a 1 mana card. This will cause a board filling
+          // issue only when this card is Rain of Frogs
+
+          // The remaining playable cards are Green Prototypes, Summon Militia, Toxic Sacrifice and Rain of Frogs
+          // In these cases only the emptyCellsIndicator variable represents how many cells are free
+          // In the other cases it is not needed and will never be set to 0
+
+          const { emptyCellsIndicator } = newState.specifics
+
+          switch (id) {
```

Comment:
I’m not sure I understand why we check anything else than Rain of Frogs?

Reply:
Rain of Frogs the TS should allow Green Prototypes to be played